tests/e2e/tests/pilot/authn_policy_test.go

```diff
@@ -67,3 +69,76 @@ func TestAuthNPolicy(t *testing.T) {
 		}
 	})
 }
+
+func TestAuthNJwt(t *testing.T) {
+	// V1alpha3 == true implies envoyv2, jwt authn doesn't work for v2 so skip it now.
+	if tc.V1alpha3 {
+		t.Skipf("Skipping %s: V1alpha3=true", t.Name())
+	}
+
+	// JWT token used is borrowed from https://github.com/istio/proxy/blob/master/src/envoy/http/jwt_auth/sample/correct_jwt.
+	// The Token expires in year 2132, issuer is 628645741881-noabiu23f5a8m8ovd8ucv698lj78vv0l@developer.gserviceaccount.com.
+	// Test will fail if this service account is deleted.
+	p := "testdata/authn/v1alpha1/correct_jwt"
+	token, err := ioutil.ReadFile(p)
+	if err != nil {
+		t.Fatalf("failed to read %q", p)
+	}
+	validJwtToken := string(token)
+
+	cfgs := &deployableConfig{
+		Namespace:  tc.Kube.Namespace,
+		YamlFiles:  []string{"testdata/authn/v1alpha1/authn-policy-jwt.yaml.tmpl"},
+		kubeconfig: tc.Kube.KubeConfig,
+	}
+	if err := cfgs.Setup(); err != nil {
+		t.Fatal(err)
+	}
+	defer cfgs.Teardown()
+
+	cases := []struct {
+		dst          string
+		src          string
+		path         string
+		port         string
+		validToken   string
+		invalidToken string
+		expect       string
+	}{
+		{dst: "a", src: "b", path: "/", port: "", validToken: "", invalidToken: "", expect: "200"},
+		{dst: "a", src: "c", path: "/xyz", port: "80", validToken: "", invalidToken: "", expect: "200"},
+
+		{dst: "b", src: "a", path: "/", port: "", validToken: "", invalidToken: "", expect: "200"},
+		{dst: "b", src: "a", path: "/xyz", port: "80", validToken: "", invalidToken: "", expect: "200"},
+		{dst: "b", src: "c", path: "/", port: "", validToken: validJwtToken, invalidToken: "", expect: "200"},
+		{dst: "b", src: "d", path: "/xyz", port: "8080", validToken: "", invalidToken: "testToken", expect: "200"},
+
+		{dst: "c", src: "a", path: "/", port: "80", validToken: validJwtToken, invalidToken: "", expect: "200"},
+		{dst: "c", src: "a", path: "/xyz", port: "8080", validToken: "", invalidToken: "invalidToken", expect: "401"},
```

Comment:
This is strange. Policy should apply on port 80 only, why 8080 returns 401.

Reply:
tried using cmd and got same result, looks like a bug somewhere..

Reply:
Can you debug that. It seems wrong

Reply:
turn out it's test code bug (missing port when constructing url), sorry for the confusion.